substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -495,3 +491,46 @@ impl ConfigDef {
 		})
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	#[test]
```

Comment:
These are good, but an ultimate way to test these are UI tests. we should already have a few around this particular item, which you can now extend to check the new logic as well.

Reply:
(maybe the existing UI tests already cover this entirely)

Reply:
@kianenigma I have added [tests/pallet_ui/pass/event_type_bound_system_config_assoc_type.rs](https://github.com/paritytech/polkadot-sdk/pull/1474/files#diff-8cb9c24184fafb43d4ac8096318ba8b5e7d281445e70065747a5ff9c9b938360) which checks for the `frame_system::Config<...>` case. Most of the tests under `tests/pallet_ui/pass` already check for the general case `frame_system::Config`. `trait_no_supertrait` and `event_type_invalid_bound` check for the compilation fail.
But testing the `frame::deps::frame_system...` cases would require a frame crate. I would suggest to either test that branch once we actually have the frame crate, or alternatively that we remove that conditional branch for now to add it back in the next PR.
WDYT?